doc: changelog.md updates

Totktonada opened the pull request:
* Fixed a typo in a link anchor.
* Added an alternative process (to better highlight why we stick with our one).

igormunkin reviewed (approved):
@Totktonada, thanks for the changes! LGTM since they are trivial.

Side note: I believe we can fix errors and typos, when building the final version of the changelog at the moment git tag is set, but yes, our approach is more flexible and convenient (at least for me).

kyukhin commented on doc/changelogs.md:
```diff
@@ -151,3 +151,16 @@ will be most convenient for our users.
 [gitlab_changelog]: https://docs.gitlab.com/ee/development/changelog.html
+
+## Alternatives
+
```
I see no reason to use such an approach. Especially after cooking few final logs. If we use entries multiple times and it contains typos (which is frequently the case), then we need to fix the final draft multiple times.

Please, provide some reasoning here.

Totktonada replied:
I don't propose to replace our current process (I think it is the best for us, no reason to change). I just want to mention all possible alternatives.

kyukhin replied:
I think this is not an alternative due to the reasons I've mentioned above.

Totktonada replied:
How should I call the section to don't make it confusing? Alternatives considered?

kyukhin replied:
I think we should not mention wrong ways.

Totktonada replied:
Okay, thanks for the opinion. I still think that we should not only describe 'how' we doing things, but also 'why'. That's the common principle in development. The example from our [commit message suggestions](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message):

> 8. Use the body to explain **what and why** vs. how.

Totktonada replied:
I tried to make it as explicit as possible that this alternative is not a proposal to use it. See the list of changes below.

Totktonada replied:
I think that with given precautions it should look good. I'll resolve this thread.

Totktonada commented:
* Renamed 'Alternatives' to 'Alternatives considered'.
* Added the explicit list of considered alternatives (reasons 'why not' are still in the 'Reasoning' section).
* Highlighted the problems we would have without ability to modify a changelog entry at any time (the scenario is suggested by @kyukhin).
* In the commit message:
  - Replaced 'a possible alternative' with 'a declined alternative'.
  - Added 'that' for grammar.
* Rebased on current master.

kyukhin commented on doc/changelogs.md:
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
+work for us, but there are several downsides:
+
+* A developer should remember about the changelog when writting a commit
+  message.
```
Typo. writting -> writing.

Totktonada replied:
Thanks! Fixed.

kyukhin commented on doc/changelogs.md:
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
+  message.
+* There is no ability to modify a changelog entry if a mistake is found later.
+
+It is easy to forget to fix a mistake if we can do it only just before a
```
I'd change wording: "... if we can only do it ...".

Totktonada replied:
I rewrote it entirely.

kyukhin commented on doc/changelogs.md:
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
+
+It is easy to forget to fix a mistake if we can do it only just before a
+release. The ability to modify a changelog entry becomes even more important,
+when the same changelog entry spread across different release branches and
```
spread -> spreads. Also, consider comma before "and".

Totktonada replied:
I rewrote this sentence entirely.

kyukhin commented on doc/changelogs.md:
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
+More alternatives are below.
+
+### Store a changelog entry in a commit message
+
```
Okay, looks like you are obsessed with mentioning this useless method. OK, let's add it with mentioned precautions.

Totktonada replied:
> Prior to stick with current process we considered several alternatives.

> See the previous section for reasons why we didn't pick up something from this
> list.

> It should mostly work for us, but there are several downsides:

Is it enough?

kyukhin replied:
I'd state explicitly: "We'd avoid using this approach due to the following drawbacks:"

Totktonada replied:
Applied:

```diff
-This process is [used in librepo][librepo_changelog_process]. It should mostly
-work for us, but there are several downsides:
+This process is [used in librepo][librepo_changelog_process]. We'd avoid using
+this approach due to the following drawbacks:
```

kyukhin commented on doc/changelogs.md:
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
+* There is no ability to modify a changelog entry if a mistake is found later.
+
+It is easy to forget to fix a mistake if we can do it only just before a
+release. The ability to modify a changelog entry becomes even more important,
```
The comma is useless.

Totktonada replied:
Seems so. Anyway, I don't satisfied by this wording and made another attempt to rewrite it.
